backend/fms_core/services/sample_next_step.py

```diff
@@ -48,4 +48,125 @@ def queue_sample_to_study_workflow(sample_obj: Sample, study_obj: Study, order:
+    """
+    Deletes a SampleNextStep instance to indicate the removal of a sample from a workflow at a specific step. 
+    Args:
+        `sample_obj`: Sample instance to be dequeued from the study workflow.
+        `study_obj`: Study instance currently associated to the sample.
+        `order`: Positive integer indicating the current position of the sample in the workflow.
+                 
+    Returns:
+        Tuple containing the number of SampleNextStep instances deleted, the error messages and the warning messages. 
+
+    """
+    dequeued = None
+    errors = []
+    warnings = []
+
+    if not isinstance(sample_obj, Sample):
+        sample_obj = None
+        errors.append(f"A valid sample instance must be provided.")
+
+    if not isinstance(study_obj, Study):
+        study_obj = None
+        errors.append(f"A valid study instance must be provided.")
+    
+    if not order:
+        order = None
+        errors.append(f"A step order must be provided.  ")
+    
+    step_order=None
+    # Dequeuing from a specific step
+    if order is not None:
+        try:
+            step_order = StepOrder.objects.get(order=order, workflow=study_obj.workflow)
+        except StepOrder.DoesNotExist:
+            errors.append(f"No step found for the given order.")
+    
+    # Dequeuing from study workflow by deleting the SampleNextStep instance
+    if sample_obj and study_obj and step_order and not errors:
+        try:
+            num_deleted, _ = SampleNextStep.objects.filter(sample=sample_obj, study=study_obj, step_order=step_order).delete()
+        except Exception as err:
+            errors.append(err)
+    return num_deleted, errors, warnings
+
+def dequeue_sample_from_all_steps_study_workflow(sample_obj: Sample, study_obj: Study) -> Tuple[Union[int, None], List[str], List[str]]:
+    """
+    Deletes a SampleNextStep instance to indicate the removal of a sample from a specified workflow at any step. 
+    Meaning that if the sample is queued in 2 different steps in the provided study, both instances will be deleted. 
+    Args:
+        `sample_obj`: Sample instance to be dequeued from the study workflow.
+        `study_obj`: Study instance currently associated to the sample.
+                 
+    Returns:
+        Tuple containing the number of SampleNextStep instances deleted, the error messages and the warning messages. 
+
+    """
+    dequeued = None
+    errors = []
+    warnings = []
+
+    if not isinstance(sample_obj, Sample):
+        sample_obj = None
+        errors.append(f"A valid sample instance must be provided.")
+
+    if not isinstance(study_obj, Study):
+        study_obj = None
+        errors.append(f"A valid study instance must be provided.")
+    
+    # Dequeuing from study workflow by deleting the SampleNextStep instance
+    if sample_obj and study_obj and not errors:
+        try:
+            num_deleted, _ = SampleNextStep.objects.filter(sample=sample_obj, study=study_obj).delete()
```

Comment:
This is a batch delete() this will not trigger our revision update. To use the model delete (instead of the manager delete) you would have to iterate on instances.